Add component-base repo

nikhita opened the pull request:
- Repo creation request: https://github.com/kubernetes/org/issues/330
- Repo with initial empty commit: https://github.com/kubernetes/component-base
- PR against test-infra for branch protection: https://github.com/kubernetes/test-infra/pull/10635
- PR for staging against k/k: https://github.com/kubernetes/kubernetes/pull/72569
- Dependencies: As per PR [`Godeps.json`](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/component-base/Godeps/Godeps.json),
  -  only apimachinery is a dependency.
  - `component-base` is a dependency to `kube-proxy`, `kube-scheduler` and `kube-controller-manager`.
  - it is safe to add future dependencies per https://github.com/kubernetes/publishing-bot/pull/142#discussion_r245485117 as well, so they have been added too.


/cc @luxas @sttts 
/assign @sttts 

/hold 

I'll consider it safe to cancel the hold when:
- [x] The published repo has been created and has an initial empty commit: https://github.com/kubernetes/component-base.
- [x] The PR against staging (https://github.com/kubernetes/kubernetes/pull/72569) has merged.
- [x] Dependencies are verified after the above PR is merged (just to double check). Note: future dependencies have also been added.
- [x] The PR against test-infra for branch protection is merged (https://github.com/kubernetes/test-infra/pull/10635)

sttts commented on configs/kubernetes-rules-configmap.yaml:
```diff
@@ -1095,6 +1095,8 @@ data:
         dependencies:
         - repository: apimachinery
           branch: master
+        - repository: component-base
```
also apiserver and the different sample-* projects will depend on it.

nikhita replied:
> will depend on it.

Is it safe to add component-base as a dependency to those repos even if it is not a dependency yet? For now, only kube-proxy, kube-scheduler and kube-controller-manager have it as a dependency (as per https://github.com/kubernetes/kubernetes/pull/72569)

sttts replied:
it only means that the order of publishing is accordingly. It does not change anything else.

luxas replied:
Great, then lets do it

nikhita replied:
:heavy_check_mark: Done.

sttts commented on configs/kubernetes-rules-configmap.yaml:
```diff
@@ -1269,3 +1275,13 @@ data:
+          dir: staging/src/k8s.io/component-base
+        name: master
+        dependencies:
+        - repository: apimachinery
```
it also depends on client-go and api.

nikhita replied:
Same question as above ^, right now in the PR component-base only depends on apimachinery: https://github.com/kubernetes/kubernetes/blob/66d879c94637489c03d16651ba01a15e60b6116a/staging/src/k8s.io/component-base/Godeps/Godeps.json.

luxas replied:
@sttts shall we wait until it actually depends on those two as well? or is this just (again) about publishing order?

nikhita replied:
I have added here as well, but let me know if you'd like to wait.

> or is this just (again) about publishing order?

Looking at the code, it looks like the list of dependencies are just used to remove the values for those deps in `Godeps/Godeps.json`, and then regenerate them. If it doesn't find a dep, it'll be a no-op. 

https://github.com/kubernetes/publishing-bot/blob/54cd0ea15d70bd75141fc02405f2d8203fd3fbfa/artifacts/scripts/util.sh#L815-L819

But let's wait for @sttts to confirm this. :)

sttts replied:
see https://github.com/kubernetes/publishing-bot/pull/142#discussion_r245485117

luxas commented on hack/fetch-all-latest-and-push.sh:
```diff
@@ -33,26 +33,27 @@ if [ "$#" -ge 2 ]; then
 fi
 GITHUB_HOST=${GITHUB_HOST:-github.com}
 repos=(
-    apimachinery
```
Thanks for sorting!

luxas commented:
/assign

luxas commented:
@nikhita both https://github.com/kubernetes/kubernetes/pull/72569 and https://github.com/kubernetes/test-infra/pull/10635 have now merged, I think we should be good to go here :)

nikhita commented:
Updated the check list in the main PR body and cancelling the hold. :rocket: 

/hold cancel

@sttts can you approve?

nikhita commented:
I updated to make sure that `component-base` comes before `apiserver` (because apiserver depends on it, so it should come first in the list). And I also added `component-base` as a dependency to `kube-aggregator` because `kube-aggregator` _will_ depend on it later.

@luxas @sttts  Ptal.

sttts commented:
/lgtm
/approve

/hold

luxas commented:
Why hold? Do you need some bugfix(es) to the bot merged first?

luxas commented:
ah I see

nikhita commented:
Closing in favour of https://github.com/kubernetes/publishing-bot/pull/144. These commits have been picked in https://github.com/kubernetes/publishing-bot/pull/144.
